python/ray/data/read_api.py

```diff
@@ -1036,6 +1036,18 @@ def read_json(
         >>> ds.take(1)
         [{'order_number': 10107, 'quantity': 30, 'year': '2022', 'month': '09'}]
 
+        When reading large files, the default block size configured in PyArrow can be too small,
+        resulting in the following error:
+        ``pyarrow.lib.ArrowInvalid: straddling object straddles two block boundaries
+        (try to increase block size?)``
+
+        To resolve this, use the ``read_options`` parameter to set a larger block size:
+
+        >>> import pyarrow.json as pajson
+        >>> block_size = 10 << 20 # Set block size to 10MB
+        >>> ray.data.read_json(...,
+        ...     read_options=pajson.ReadOptions(block_size=block_size))
```

Comment:
```suggestion
        >>> ray.data.read_json(
        ...     s3://anonymous@ray-example-data/log.json,
        ...     read_options=pajson.ReadOptions(block_size=block_size)
        ... )
```

Reply:
If you wanted to, you could add a `large-log.json` to make this example more realistic, but I don't think that's necessary